### DISCUSSION TITLE
Binding Entities near the end of Chapter 10

### DISCUSSION THREAD

**Discussion by MPAGenie** [symfony/symfony#50711] [Symfony Book: The Fast Track]
I've been working through the book recently, and once it gets into the "pagination" section, a strange bug seems to appear on my local - 

`Binding entities to query parameters only allowed for entities that have an identifier.
Class "App\Entity\Conference" does not have an identifier.`

I've followed the book to the letter and been unable to work out why this was happening, until a random stack overflow thread I came across made note that this version of the book is for older versions of Symfony, and has a package missing, now long unsupported, that causes this bug. Any ideas?

**Comment by yun548** [comment]
Your class needs a [unique identifier](https://www.doctrine-project.org/projects/doctrine-orm/en/2.15/reference/basic-mapping.html#identifiers-primary-keys) so Doctrine can identify each record and paginate accordingly.

**Reply by MPAGenie** [reply]
On what part? can you give a little more context? I'm still pretty new to all this

**Comment by stof** [comment]
The Conference entity created in chapter 8 _has_ an identifier field (named `id`). So to me, getting this error makes me think that your code does not actually correspond to following the book to the letter.

**Reply by MPAGenie** [reply]
My Conference entity controller has ...
`#[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;`
    
    On lines 13-16. is that not correct?

**Reply by yun548** [reply]
This should not be in your Controller but in the `Conference` entity you generated using the command.

The generated entity should look like this : 

```
namespace App\Entity;

use App\Repository\ConferenceRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ConferenceRepository::class)]
class Conference
{
    #[ORM\Column(type: 'integer')]
    #[ORM\Id, ORM\GeneratedValue()]
    private $id;

    #[ORM\Column(type: 'string', length: 255)]
    private $city;

    // ...

    public function getCity(): ?string
    {
        return $this->city;
    }

    public function setCity(string $city): self
    {
        $this->city = $city;

        return $this;
    }

    // ...
}
```

I suggest you go over the [database chapter](https://symfony.com/doc/6.2/the-fast-track/fr/8-doctrine.html#configurer-doctrine-orm) one more time to make sure everything is in order there
